chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -852,7 +853,26 @@ impl PeerManagerActor {
                     None
                 }
             })
-            .collect::<Vec<_>>()
+            .collect::<Vec<_>>();
+
+        // remove peers to which we have no route from the list
+        result.retain(|peer| {
+             if self
+                .routing_table_view
+                .peer_forwarding
+                .get(&peer.peer_info.id)
+                .unwrap_or(&Vec::new())
+                .is_empty()
+            {
+                error!(target: "network", "returning peer to ClientActor to which we have no route: {}", &peer.peer_info.id);
+                false
+            } else {true}
+        });
+        if result.is_empty() {
+            error!(target: "network", "result of highest_height_peers became empty");
```

Comment:
maybe also explain in this log statement what could be the possible reason?

Reply:
@mm-near I suspect that there is an issue. We return in this function a peer, to which we don't have a route.
My plan is to do two things
- add a print, and see if that's really the same. once we confirm that this is the case. 
a)Once we confirm that, we can dedicate some time to add extra tests to find out where is the bug in logic that causes us to choose peers out of sync.
b) if that's not the case, we will have to look at a bug, in another code path
- There is no need to send list of peers to which we don't have path to, they would be dropped anyway. That's why I added `retain`.


Right now I'm working on other changes. This is a simple change, that should let us know whenever the issue is somewhere in this code.